velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -268,27 +270,165 @@ TpchPlan TpchQueryBuilder::getQ6Plan() const {
   return context;
 }
 
-const std::vector<std::string> TpchQueryBuilder::kTableNames_ = {"lineitem"};
+TpchPlan TpchQueryBuilder::getQ18Plan() const {
+  static const std::vector<std::string> kTableNames = {
+      "lineitem", "orders", "customer"};
+  std::vector<std::string> lineitemSelectedColumns = {
+      "l_orderkey", "l_quantity"};
+  std::vector<std::string> ordersSelectedColumns = {
+      "o_orderkey", "o_custkey", "o_orderdate", "o_totalprice"};
+  std::vector<std::string> customerSelectedColumns = {"c_name", "c_custkey"};
+
+  // Get the corresponding selected column names in the file.
+  const auto lineitemAliasSelectedColumns =
+      getColumnAliases(kTableNames[0], lineitemSelectedColumns);
+  const auto ordersAliasSelectedColumns =
+      getColumnAliases(kTableNames[1], ordersSelectedColumns);
+  const auto customerAliasSelectedColumns =
+      getColumnAliases(kTableNames[2], customerSelectedColumns);
+
+  auto lineitemColumnSelector =
+      std::make_shared<facebook::velox::dwio::common::ColumnSelector>(
+          tableMetadata_.at(kTableNames[0]).type, lineitemAliasSelectedColumns);
+  auto ordersColumnSelector =
+      std::make_shared<facebook::velox::dwio::common::ColumnSelector>(
+          tableMetadata_.at(kTableNames[1]).type, ordersAliasSelectedColumns);
+  auto customerColumnSelector =
+      std::make_shared<facebook::velox::dwio::common::ColumnSelector>(
+          tableMetadata_.at(kTableNames[2]).type, customerAliasSelectedColumns);
+
+  auto lineitemSelectedRowType =
+      lineitemColumnSelector->buildSelectedReordered();
```

Comment:
consider introducing a helper function that takes a list of column names and returns a RowType

```
auto lineitemRowType = getRowType(kLineitem, lineitemColumns)
```